<a href="https://colab.research.google.com/github/AbbisreeSaadhvi/Python-Projects/blob/main/Hypothesis_Testing.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Hypothesis Testing Examples with Explanations

##1. Z-Score Test (One-Sample Z-Test)

**Scenario:**
A company claims that the average weight of their product is 500 grams. A sample of 30 products has an average weight of 490 grams with a standard deviation of 20 grams. We want to test if the average weight is significantly different from 500 grams at a 0.05 significance level.

**Hypothesis:**

Null Hypothesis (𝐻0): 𝜇=500

Alternative Hypothesis (𝐻1): 𝜇≠500

In [1]:
import numpy as np
from scipy.stats import norm

# Given data
sample_mean = 490
population_mean = 500
std_dev = 20
n = 30

# Z-score calculation
z_score = (sample_mean - population_mean) / (std_dev / np.sqrt(n))
p_value = 2 * (1 - norm.cdf(abs(z_score)))

z_score, p_value

(-2.7386127875258306, 0.0061698993205441255)

**Interpretation:**
The p-value (0.0062) is less than the significance level (0.05), so we reject the null hypothesis. There is significant evidence that the average weight of the product is different from 500 grams.

##2. Chi-Square Test

**Scenario:**
A survey is conducted to determine if there is a preference for three brands of a product (A, B, and C) among 300 consumers. The observed frequencies are: Brand A - 120, Brand B - 90, Brand C - 90. We want to test if the preferences are equally distributed at a 0.05 significance level.

**Hypothesis:**

Null Hypothesis (𝐻0): Preferences are equally distributed.

Alternative Hypothesis (𝐻1): Preferences are not equally distributed.

In [4]:
import numpy as np
from scipy.stats import chi2_contingency

# Observed data
observed = np.array([120, 90, 90])

# Expected data
expected = np.array([100, 100, 100])

# Chi-Square test
chi2, p, dof, expected = chi2_contingency([observed, expected])
chi2, p

(2.8708133971291865, 0.23801854252784904)

**Interpretation:**

The p-value is 0.2380, which is greater than the significance level of 0.05. Since the p-value is greater than 0.05, we fail to reject the null hypothesis.

There is not enough evidence to conclude that the preferences for Brand A, Brand B, and Brand C are significantly different. This means that the survey results do not provide sufficient evidence to suggest that consumers prefer one brand over the others, and the preferences appear to be equally distributed among the three brands.

##3. ANOVA Test

**Scenario:**

We have three groups of students with test scores from three different teaching methods. We want to determine if there are significant differences in the mean test scores among the three groups at a 0.05 significance level.

**Hypothesis:**

Null Hypothesis (𝐻0): All group means are equal.

Alternative Hypothesis (𝐻1): At least one group mean is different.

In [7]:
import numpy as np
from scipy import stats

# Sample data
group1 = np.random.normal(70, 10, 30)
group2 = np.random.normal(75, 10, 30)
group3 = np.random.normal(65, 10, 30)

# Perform ANOVA test
f_stat, p_value = stats.f_oneway(group1, group2, group3)
f_stat, p_value

(7.213843794800864, 0.001262637496502434)

**Interpretation:**

The p-value (0.0012) is less than the significance level (0.05), so we reject the null hypothesis. There is significant evidence that at least one group mean is different from the others.

#Summary


**Z-Score Test:**

Used to determine if a sample mean significantly differs from a known population mean.

**Chi-Square Test:**

Used to test if there is a significant association between categorical variables or if observed frequencies differ from expected frequencies.

**ANOVA Test:**

Used to compare the means of three or more groups to see if at least one group mean is significantly different from the others.